third_party/move/move-compiler-v2/src/env_pipeline/lambda_lifter.rs

```diff
@@ -79,12 +79,7 @@ pub struct LambdaLiftingOptions {
 pub fn lift_lambdas(options: LambdaLiftingOptions, env: &mut GlobalEnv) {
     // Go over target modules one by one. Since in each iteration
     // we need to mutate the module, iterate over a vector of plain ids.
-    for module_id in env
-        .get_target_modules()
-        .into_iter()
-        .map(|me| me.get_id())
-        .collect_vec()
-    {
+    for module_id in env.get_modules().map(|me| me.get_id()).collect_vec() {
```

Comment:
nit: the comment still says "target modules"

Reply:
Ah, already spotted by @vineethk 👍

Reply:
I think we should do what is said in the PR description, do it like the bytecode generator, and actually go over the target modules _and only_ the non-target modules which are transitively used by all those modules. The algorithm is [here](https://github.com/aptos-labs/aptos-core/blob/8d630a3f06c84dee350bf4d32f36ba957a1e2ab3/third_party/move/move-compiler-v2/src/lib.rs#L252). You can factor it into `model.rs`, as a function of`GlobalEnv`, like `get_target_modules_transitive_closure`.

Reply:
PS. You don't need the by-function treatment like in the code, but can just use `get_used_modules` to compute the closure.

Reply:
done